File: Client/src/app/core/models/signal-r/update-folder-ws.ts
import { PreviewNotesInFolder } from 'src/app/content/folders/models/preview-notes-in-folder.model';
import { MergeTransaction } from 'src/app/content/notes/full-note/content-editor/text/rga/types';
import { PositionEntityModel } from 'src/app/content/notes/models/position-note.model';

export interface UpdateFolderWS {
  folderId: string;
  color: string;
  titleTransactions: MergeTransaction<string>[];
  isUpdateTitle: string;
  previewNotes: PreviewNotesInFolder[];
  idsToRemove: string[];
  idsToAdd: string[];
  positions: PositionEntityModel[];
}

File: Backend/NootsWEB/Common/DTO/WebSockets/UpdateFolderWS.cs
using Common.DTO.Folders;
using Common.DTO.Notes;
using Noots.RGA_CRDT;
using System;
using System.Collections.Generic;

namespace Common.DTO.WebSockets
{
    public class UpdateFolderWS
    {
        public Guid FolderId { set; get; }

        public string Color { set; get; }

        public List<MergeTransaction<string>> TitleTransactions { set; get; }

        public bool IsUpdateTitle { set; get; }

        public List<EntityPositionDTO> Positions { set; get; } = new();

        public List<Guid> IdsToAdd { set; get; } = new();

        public List<Guid> IdsToRemove { private set; get; } = new();

        public List<NotePreviewInFolder> PreviewNotes { set; get; } = new();
    }
}
